media-sound/zynaddsubfx/zynaddsubfx-3.0.5-r1.ebuild

```diff
@@ -0,0 +1,96 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake flag-o-matic
+
+DESCRIPTION="Software synthesizer capable of making a countless number of instruments"
+HOMEPAGE="http://zynaddsubfx.sourceforge.net/"
+SRC_URI="mirror://sourceforge/zynaddsubfx/${P}.tar.bz2"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="alsa doc dssi +fltk jack lash portaudio zest"
+
+REQUIRED_USE="|| ( alsa jack portaudio )"
+
+BDEPEND="
+	virtual/pkgconfig
+	doc? ( app-doc/doxygen )
+"
+DEPEND="
+	>=dev-libs/mxml-2.2.1
+	media-libs/liblo
+	sci-libs/fftw:3.0
+	sys-libs/zlib
+	alsa? ( media-libs/alsa-lib )
+	dssi? ( media-libs/dssi )
+	fltk? (
+		>=x11-libs/fltk-1.3:1
+		x11-libs/libX11
+		x11-libs/libXpm
+	)
+	jack? ( virtual/jack )
+	lash? ( media-sound/lash )
+	portaudio? ( media-libs/portaudio )
+"
+RDEPEND="${DEPEND}
+	zest? ( media-sound/zyn-fusion )
+"
+
+PATCHES=( "${FILESDIR}"/${P}-docs.patch )
+
+DOCS=( ChangeLog HISTORY.txt README.adoc )
+
+src_prepare() {
+	cmake_src_prepare
+
+	if ! use dssi; then
+		sed -i -e '/pkg_search_module.*DSSI/s/^/#DONT/' src/CMakeLists.txt || die
+	fi
+	if ! use jack; then
+		sed -e '/pkg_check_modules.*JACK/s/^/#DONT/' -i {rtosc,src}/CMakeLists.txt || die
+	fi
+	if ! use lash; then
+		sed -i -e '/pkg_search_module.*LASH/s/^/#DONT/' src/CMakeLists.txt || die
+	fi
+	if ! use portaudio; then
+		sed -i -e '/pkg_check_modules.*PORTAUDIO/s/^/#DONT/' src/CMakeLists.txt || die
+	fi
+
+	# FIXME upstream: sandbox error
+	sed -i -e '/add_subdirectory(bash-completion)/d' doc/CMakeLists.txt || die
+}
+
+src_configure() {
+	append-cxxflags -std=c++11
+
+	local mycmakeargs=(
+		-DPluginLibDir=$(get_libdir)
+		$(cmake_use_find_package alsa Alsa)
+		$(cmake_use_find_package doc Doxygen)
+		$(cmake_use_find_package fltk FLTK)
+	)
+	if use zest ; then
```

Comment:
Shouldn't we have a exactly one of for `fltk` and `zest`?
Now you can have both set but you'll still always get `fltk` for the plugin UI, I think that's a bit confusing.

Reply:
I was wondering that myself when I was writing the ebuild, but then I looked at the REQUIRED_USE documentation and it says it should be used sparingly:

https://devmanual.gentoo.org/general-concepts/use-flags/index.html#conflicting-use-flags

In particular it seems exclusions like that are more useful for dependencies, while for leaf level ebuilds it makes more sense to favour user intent (which, if they ask for zest, probably means they don't want fltk).

An einfo message to this effect might be in order, though.